### PR TITLE
Openapi > VersionedModelEndPoint > Normalize & Add responses

### PR DESCRIPTION
**Pull request by flyusfly**
- [x] - VersionedModelEndpoint.get_versions() - Normalize & Add responses
- [x] - VersionedModelEndpoint.get_version() - Normalize & Add responses
- [x] - VersionedModelEndpoint.post_version() - Normalize & Add responses
- [x] - VersionedModelEndpoint.get_redirects() - Normalize & Add responses
- [x] - VersionedModelEndpoint.put_delete_redirects() > Split put and delete redirects & Normalize & Add responses
- [x] - VersionedModelEndpoint > Simplify api identifier parameters
- [x] - VersionedModelEndpoint.delete_redirets() > Delete 422 response
- [x] - Diff.get_collection() > Generic 400 response
- [x] - VersionedModelEndpoint.post_version() > body type param must be 'object'

### CODE REVIEW

**Comment by flyusfly** [comment]
Waiting for #253 PR to simplify api identifier parameters for VersionedModelEndpoints.

**Comment by flyusfly** [comment]
Is it possible to have a check on this PR, please?

**Comment by yohanboniface** [comment]
As I still see duplicated "identifier" parameter declarations, I thought it was not ready to review (as we discussed this already).

File: ban/http/api.py [hunk 1]
```diff
@@ -417,48 +448,110 @@ def post_version(self, identifier, ref):
...
+                                $ref: '#/definitions/Redirect'
+                        total:
+                            name: total
+                            type: integer
+                            description: total resources available
+            401:
+                $ref: '#/responses/401'
+            404:
+                $ref: '#/responses/404'
+            422:
```
Comment (yohanboniface) [inline ban/http/api.py]: Humm, how do we return a 422 here?

File: ban/http/api.py [hunk 2]
```diff
@@ -574,16 +667,33 @@ def get_collection(self):
...
+                        total:
+                            name: total
+                            type: integer
+                            description: total resources available
+                        collection:
+                            name: collection
+                            type: array
+                            items:
+                                $ref: '#/definitions/Diff'
+            400:
```
Comment (yohanboniface) [inline ban/http/api.py]: I think we can just point to the generic 400 reference here, the detail will be in the response body.

File: ban/http/api.py [hunk 3]
```diff
@@ -391,18 +408,32 @@ def get_version(self, identifier, ref):
...
             - $ref: '#/parameters/identifier'
             - name: ref
               in: path
-              type: string
+              type: integer
               required: true
               description: version reference, either a date or an increment.
+            - name: body
+              in: body
+              type: string
```
Comment (yohanboniface) [inline ban/http/api.py]: Now I have a new doubt: does it make sense to have "type: string" with a "schema"? Shouldn't this be "type: object" instead?

**Comment by flyusfly** [comment]
Sorry, for this previous commit.
Because body parameter is linked to a schema, type 'object' has not to be defined.

File: ban/http/api.py [hunk 4]
```diff
@@ -417,48 +447,108 @@ def post_version(self, identifier, ref):
...
+            - name: old
+              in: path
+              type: string
+              required: true
+              description: old {resource} identifier:value
         responses:
-            200:
-                description: A list of redirects.
+            204:
+                description: redirect was successfully deleted.
```
Comment (yohanboniface) [inline ban/http/api.py]: I guess this is not where you planned to add this schema definition, is it?

Comment (flyusfly) [inline ban/http/api.py]: Sorry, I don't understand.
. Lines 460 & 461 are extracted from get_redirects(). 
. Lines 503 & 504 are extracted from delete_redirects().

File: ban/http/api.py [hunk 5]
```diff
@@ -391,18 +408,31 @@ def get_version(self, identifier, ref):
...
             - $ref: '#/parameters/identifier'
             - name: ref
               in: path
-              type: string
+              type: integer
               required: true
               description: version reference, either a date or an increment.
+            - name: body
+              in: body
+              schema:
```
Comment (yohanboniface) [inline ban/http/api.py]: I think the schema is wrong here: it's pointing to the resource, while it's just a `{status: xxx}` one.

File: ban/http/api.py [hunk 6]
```diff
@@ -417,48 +447,97 @@ def post_version(self, identifier, ref):
...
               required: true
-              description: old identifier.
+              description: Old {resource} identifier:value
         responses:
-            204:
-                description: redirect was successful.
             201:
-                description: redirect was created.
+                description: redirect was successfully created.
+                schema:
```
Comment (yohanboniface) [inline ban/http/api.py]: Not sure why we have a `schema` here. The 201 response body is empty.

**Comment by jdesboeufs** [comment]
@flyusfly où en est cette PR ?

**Comment by flyusfly** [comment]
Elle attend un commit...

**Comment by jdesboeufs** [comment]
Quoi ? Qui ?

**Comment by flyusfly** [comment]
Les 2 reviews sur cette PR ont été effectuées le 07/12 (Thx Yohan).
J'ai effectué les changements en conséquence le 13/12...

Si Yohan passe par là, et veut bien faire une review et/ou commiter, je suis preneur...

Thx in advance...

**Comment by flyusfly** [comment]
Bonjour Yohan,

Pourrais-tu regarder cette PR, stp ?
Cela permettrait de clôturer ce sujet openapi.

Merci d'avance...

**Comment by jdesboeufs** [comment]
Dernière chance, après je merge ;)

**Pull request merged**